action/protocol/staking/vote_reviser.go

```diff
@@ -35,11 +37,57 @@ func (vr *VoteReviser) Revise(csm CandidateStateManager, height uint64) error {
 		if err != nil {
 			return err
 		}
+		if vr.needCorrectCands(height) {
+			if err := vr.correctAliasCands(csm, cands); err != nil {
+				return err
+			}
+			if csm, err = vr.reconstructCandCenter(csm); err != nil {
+				return err
+			}
+		}
 		vr.storeToCache(height, cands)
 	}
 	return vr.flush(height, csm)
 }
 
+func (vr *VoteReviser) correctAliasCands(csm CandidateStateManager, cands CandidateList) error {
+	_, _, owners, err := readCandCenterStateFromStateDB(csm.SM(), 0)
+	switch {
+	case errors.Cause(err) == state.ErrStateNotExist:
+		return nil
+	case err != nil:
+		return err
+	}
+
+	for _, c := range cands {
+		for _, owner := range owners {
+			if c.Owner.String() == owner.Owner.String() {
+				c.Operator = owner.Operator
+				c.Reward = owner.Reward
+				c.Name = owner.Name
+			}
+		}
+	}
+	return nil
+}
+
+func (vr *VoteReviser) reconstructCandCenter(csm CandidateStateManager) (CandidateStateManager, error) {
+	cc, err := NewCandidateCenter(nil, HasAliasOption(false))
+	if err != nil {
+		return nil, err
+	}
+	view := csm.DirtyView()
+	newCSM := candSM{
+		csm.SM(),
+		cc,
+		view.bucketPool,
+	}
+	if err := newCSM.WriteView(_protocolID, newCSM.DirtyView()); err != nil {
```

Comment:
create a new cand center (with bug fixed), and write it into the view
next time `NewCandidateStateManager` will get this correct candidate center